Updated Material Dark

revosw opened the pull request:
- The command bar menus are now more consistent with borders.
- Razor markup now has consistent background color
- Server side markup now has same blue color as strings

### Changes:
- CommandBarButton:
OLD
![](https://i.imgur.com/gzkz4Mn.png)

NEW
![](https://i.imgur.com/tUFuvZM.png)

- DropDown
OLD
![](https://i.imgur.com/ut6X7D7.png)

NEW
![](https://i.imgur.com/HvXiYIP.png)

- Razor Markup
OLD
![](https://i.imgur.com/ALkoXhY.png)

NEW
![](https://i.imgur.com/ZCmJM4F.png)

Newrad0603 commented on VSColorThemes/source.extension.vsixmanifest:
```diff
@@ -32,6 +32,6 @@
     </Assets>
     <Prerequisites>
-        <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,16.0)" DisplayName="Visual Studio core editor" />
+        <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,17.0)" DisplayName="Visual Studio core editor" />
```
This change can be undone as it's no longer needed.

Newrad0603 commented on VSColorThemes/Themes/MaterialDark.xml:
```diff
@@ -1,5 +1,5 @@
 <Themes>
-  <Theme Name="Material Dark" GUID="{d34da167-b381-433a-8877-d6451a3f9c01}">
+  <Theme Name="Material Dark" GUID="{085573d9-362f-4a88-8fbe-84c509b98120}">
```
Why is the theme guid being changed? This should generally be left alone.

revosw replied:
To be honest, I have no recollection of what happened there, it was not my intent to change the GUID. I will change it back to the original value, if that's preferred?

Newrad0603 replied:
I think you may have edited the wrong file. I still see the wrong GUID in MaterialDark, as well as a bunch of CT_INVALID entries still.

Newrad0603 commented on VSColorThemes/Themes/MaterialDark.xml:
```diff
@@ -1614,6 +1614,14 @@
         <Foreground Type="CT_RAW" Source="FF81D4FA" />
       </Color>
+      <Color Name="XML Doc Comment">
+        <Background Type="CT_INVALID" Source="00000000" />
```
Please remove all instances of CT_INVALID color entries. They are non-functional, but bloat the private registry.

Newrad0603 commented on VSColorThemes/Themes/MaterialDark.xml:
```diff
@@ -1678,118 +1686,110 @@
         <Foreground Type="CT_INVALID" Source="00000000" />
       </Color>
-      <Color Name="RoslynConflictTag">
+      <Color Name="UnderlineClassificationFormatProPack">
```
I assume a reorder happened here. Could you restore the original ordering so that the real changes are easier to see?

Newrad0603 commented on VSColorThemes/Themes/MaterialDark.xml:
```diff
@@ -7612,6 +7592,10 @@
         <Background Type="CT_INVALID" Source="00000000" />
         <Foreground Type="CT_INVALID" Source="00000000" />
       </Color>
+      <Color Name="Breakpoint - Advanced (Disabled)">
```
Once the CT_INVALID entries are removed, you'll end up with empty colors, so make sure to remove those as well. Once the empty colors are removed, you may have empty categories which should also be removed.

Newrad0603 commented:
Thanks for the screenshots! Feel free to ask questions about any feedback I left.

revosw commented:
I have now made changes that addresses your points:
* From `[15.0,17.0)` to `[15.0,16.0)` in `source.extension.vsixmanifest`
* GUID back to `{d34da167-b381-433a-8877-d6451a3f9c01}`
* Removed all instances of colors with `type="CT_INVALID"`
* Removed all empty color and category definitions

revosw commented:
Big rookie mistake of me, I have now edited the correct file, MaterialDark.xml

I also have a request, regarding the theme Sublime Material Dark. I made the theme back in the day, and I don't like the theme. I don't want to maintain it. Is it okay if it is removed, or request someone else be the maintainer?

Newrad0603 commented on VSColorThemes/Themes/MaterialDark.xml:
```diff
@@ -1000,12 +981,10 @@
         <Background Type="CT_RAW" Source="FF773800" />
-        <Foreground Type="CT_INVALID" Source="00000000" />
-      </Color>
+        </Color>
```
Overall change looks good, but mind fixing the indenting on the closing tags. Several of them got messed up. Once that's done, we'll get this merged in. Thanks!

revosw commented:
Gotem, boss 😎👉👉
I reckon the changes to ThemeRegistration and vsixmanifest are okay?
